src/Storages/StorageAzureBlob.cpp

```diff
@@ -315,58 +315,90 @@ AzureClientPtr StorageAzureBlob::createClient(StorageAzureBlob::Configuration co
 
     if (configuration.is_connection_string)
     {
+        std::unique_ptr<BlobServiceClient> blob_service_client = std::make_unique<BlobServiceClient>(BlobServiceClient::CreateFromConnectionString(configuration.connection_url));
+
+        Azure::Storage::Blobs::ListBlobContainersOptions options;
+        options.Prefix = configuration.container;
+        options.PageSizeHint = 1;
+
+        auto containers_list_response = blob_service_client->ListBlobContainers(options);
+        auto containers_list = containers_list_response.BlobContainers;
+
+        bool container_exists = false;
+        for (const auto & container : containers_list)
+        {
+            if (configuration.container == container.Name)
+            {
+                container_exists = true;
+                break;
+            }
+        }
+
         result = std::make_unique<BlobContainerClient>(BlobContainerClient::CreateFromConnectionString(configuration.connection_url, configuration.container));
-        result->CreateIfNotExists();
+
+        if (!container_exists)
+        {
+            result->CreateIfNotExists();
```

Comment:
It it's readonly case maybe it make sense to throw exception?

Reply:
Updated to pass parameter `is_insert_query` to table function (had to update all table functions, but this parameter is currently only used in azureBlobStrorage). If its not insert query and container does not exist, we throw error.